src/dmd/parse.d

```diff
@@ -1340,7 +1340,13 @@ final class Parser(AST) : Lexer
      */
     private StorageClass appendStorageClass(StorageClass storageClass, StorageClass stc)
     {
-        if ((storageClass & stc) || (storageClass & AST.STC.in_ && stc & (AST.STC.const_ | AST.STC.scope_)) || (stc & AST.STC.in_ && storageClass & (AST.STC.const_ | AST.STC.scope_)))
+        static bool isIn (StorageClass stc)
```

Comment:
I think this should be a public utility function accompanying https://github.com/dlang/dmd/blob/467bd7577c440dede67a65eb8ed82f3079bf314d/src/dmd/declaration.d#L196